src/core/thread/mle.cpp

```diff
@@ -1446,15 +1446,27 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
         VerifyOrExit(counter < OPENTHREAD_CONFIG_MLE_IP_ADDRS_TO_REGISTER);
     }
 
-    // For sleepy end device, register external multicast addresses to the parent for indirect transmission
+    // For sleepy end device, register external multicast addresses to the parent for indirect transmission.
+    // Since Thread 1.2, non-sleepy MED would depend on its Thread 1.2 parent as the proxy to perform
+    // Multicast Listener Report, it registers external multicast addresses of scope larger than realm
+    // local as well.
+#if (OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_1)
     if (!IsRxOnWhenIdle())
+#endif
     {
         uint8_t      iterator = 0;
         Ip6::Address address;
 
         // append external multicast address
         while (Get<ThreadNetif>().GetNextExternalMulticast(iterator, address) == OT_ERROR_NONE)
         {
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+            // For Thread 1.2 MED, skip multicast address with scope not larger than realm local.
+            if (IsRxOnWhenIdle() && !address.IsMulticastLargerThanRealmLocal())
```

Comment:
Should we also have a check for Thread 1.1 vs 1.2 parent?

Reply:
According to Spec 5.24.3.3, if an MTD attaches to a 1.1 parent, it should perform both actions:

- Registering with MLR.req 
- Registering with MLE Address Registration TLV

Thoughts @jwhui @librasungirl

Reply:
@simonlingoogle  
 I am trying to split into small relative independent PRs to ease the review process. 
MLR would be in a separate PR. Please only focus on the purpose as the subject reveals. Thanks.

Reply:
@jwhui added parent version check. thanks.